src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -1271,6 +1271,11 @@ private bool PrepareMessageHeaders(Message message)
                             {
                                 // specifying a range doesn't make sense in the context of WCF
                             }
+                            else if (string.Compare(name, "authorization", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                // allows adding authorisation that doesnt conform to header standards, eg. AWS Signiture 4 headers
+                                _httpRequestMessage.Headers.TryAddWithoutValidation(name, value);
```

Comment:
On .NET Framework, we just blindly add any headers that we don't explicitly want to handle, which is equivalent to calling TryAddWithoutValidation. I think it you wish to add a header which doesn't conform to the HTTP spec (such as in this case), who are we to argue. So instead of this change, I think it would be better to change the code a few lines below to use TryAddWithoutValidation instead of Add.